.github/workflows/dbld-images.yml

```diff
@@ -55,18 +50,14 @@ jobs:
           gh_export UPLOAD_IMAGES_INTERNAL
 
       - name: Log in to the Container registry
-        if: env.UPLOAD_IMAGES_INTERNAL == 'true'
+        if: always() && env.UPLOAD_IMAGES_INTERNAL == 'true'
         uses: docker/login-action@v1
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Push the images
-        if: env.UPLOAD_IMAGES_INTERNAL == 'true'
+        if: always() && env.UPLOAD_IMAGES_INTERNAL == 'true'
         run: |
-          IMAGES=$(docker image ls --filter "reference=${{ env.CONTAINER_REGISTRY }}/dbld-*:latest" --format '{{.Repository}}:{{.Tag}}')
-          for IMAGE in $IMAGES; do
-            echo "Pushing image: $IMAGE"
-            docker push $IMAGE
-          done
+          dbld/rules push-images -j $(nproc) --output-sync
```

Comment:
previously this only pushed built images, while now it pushes all images ./dbld/rules knows about. what happens if a build does not finish successfully?

It looks good to me otherwise.

Reply:
Yes, good point, I was just adding this quickly in a hurry.

How about a cheap solution: adding `--keep-going` to dbld/rules?
It will make the upload step failed, but I don't know if that's an issue since the image build already failed too.

I don't think `push-images` should do a filtering for existing images internally, or using the previous `docker image ls ` filtering either.

Reply:
I like the --keep-going option. that would upload those that succeeded and the job would still fail, but I think that is fair.